airbyte-ci/connectors/connectors_insights/src/connectors_insights/hacks.py

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+from __future__ import annotations
+
+import json
+from typing import TYPE_CHECKING
+
+import requests
+from bs4 import BeautifulSoup
+from connectors_insights.consts import TEST_SUMMARY_ROOT_URL
+from google.cloud import storage
+
+if TYPE_CHECKING:
```

Comment:
Why not just always import instead?

Reply:
I can answer this one. The `if TYPE_CHECKING` block has become a best practice (including lint rules) for any imports that are not needed at runtime, aka imports that are needed for type hints _only_.

The benefits of this are twofold:

1. Avoids the perf hit of running those imports at runtime if they are not _needed_ at runtime.
2. Avoid failures from circular references - when a "full" import would otherwise circular reference. This allows sibling modules to use type hints from each other, without that turning into a circular import loop at execution time.

Reply:
I probably should have just linked this rule. It explains better than I can 😄 

https://docs.astral.sh/ruff/rules/typing-only-first-party-import/#example